src/sql/src/plan/query.rs

```diff
@@ -905,6 +905,52 @@ pub fn plan_secret_as(
     Ok(expr)
 }
 
+/// Plans an expression in the VALIDATE USING position of a `CREATE SOURCE ... FROM WEBHOOK`.
+pub fn plan_webhook_validate_using(
+    scx: &StatementContext,
+    mut expr: Expr<Aug>,
+) -> Result<MirScalarExpr, PlanError> {
+    let qcx = QueryContext::root(scx, QueryLifetime::OneShot(scx.pcx()?));
+
+    // We _always_ provide the body of the request as bytes and include the headers when validating
```

Comment:
Why? Would this policy confuse users who are using JSON? Then again, maybe we should have a policy of doing as little as possible cpu and memory wise before validating the request.

Reply:
Always using the raw bytes for validation is necessary because AFAICT all webhooks hash the original body. If we first converted it to JSON or TEXT and provided that to validation, it might not work because there's a chance our converted type != the original at a byte level, and thus the hash wouldn't compare